File: src/main/java/lesson10/Road.java
package lesson10;

public class Road extends Stage {
    public Road(int length) {
        this.length = length;
        this.description = "Дорога " + length + " метров";
    }

    @Override
    public void overcome(Car c) {
        try {
            System.out.println(c.getName() + " начал этап: " + description + "     " + System.currentTimeMillis());
            Thread.sleep(length / c.getSpeed() * 1000);
            System.out.println(c.getName() + " закончил этап: " + description + "     " + System.currentTimeMillis());
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/lesson10_kt/Road.kt
package lesson10_kt

class Road(length: Int) : Stage() {
    init {
        this.length = length
        this.description = "Дорога $length метров"
    }

    override fun overcome(c: Car) {
        try {
            println(c.name + " начал этап: " + description)
            Thread.sleep((length / c.speed * 1000).toLong())
            println(c.name + " закончил этап: " + description)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}